File: p-2/solution.rb
def fib_total(max)
  total = 0
  a, b = 0, 1
  while a < max
    if a % 2 == 0 
      total += a 
    end 
    a, b = b, a + b
  end 
  total
end 

p fib_total(4_000_000)

File: p-2/solution.js
function fibTotal(max) {
  let a = 0
  let b = 1
  let total = 0

  while( a < max ) {
    if( a % 2 == 0 ) {
      total += a
    }
    let hold = a
    a = b
    b = hold + b
  }
  return total
}

console.log( fibTotal(4_000_000) )